packages/core/entity/FileDetail/index.ts

```diff
@@ -126,56 +144,74 @@ export default class FileDetail {
 
         // AICS FMS files have paths like this in fileDetail.file_path:
         // staging.files.allencell.org/130/b23/bfe/117/2a4/71b/746/002/064/db4/1a/danny_int_test_4.txt
-        if (typeof path === "string" && path.startsWith(AICS_FMS_S3_BUCKET)) {
+        if (
+            typeof path === "string" &&
+            path.startsWith(AICS_FMS_S3_BUCKETS[this.env]) &&
+            AICS_FMS_S3_BUCKETS[this.env]
+        ) {
             return FileDetail.convertAicsS3PathToHttpUrl(path) as string;
         }
 
         // Otherwise just return the path as is and hope for the best
         return path as string;
     }
 
+    public get cloudPath(): string {
+        //// REMOVE THIS (BACKWARDS COMPAT)
+        if (this.path.startsWith("/allen")) {
+            return this.convertAicsDrivePathToAicsS3Path(this.path);
+        }
+        ////
+
+        // AICS FMS files' paths are cloud paths
+        return this.path;
+    }
+
     public get localPath(): string | null {
-        // REMOVE THIS (BACKWARDS COMPAT)
+        //// REMOVE THIS (BACKWARDS COMPAT)
         if (this.path.startsWith("/allen")) {
             return this.path;
         }
+        ////
 
-        const localPath = this.getFirstAnnotationValue("Local File Path");
-        if (localPath === undefined) {
-            return null;
+        if (this.getAnnotation("Cache Eviction Date")) {
+            return this.getLocalPath();
         }
-        return localPath as string;
+        return null;
     }
 
-    public get cloudPath(): string {
-        // REMOVE THIS (BACKWARDS COMPAT)
-        if (this.path.startsWith("/allen")) {
-            return FileDetail.convertAicsDrivePathToAicsS3Path(this.path);
+    public getLocalPath(): string | null {
+        const localPrefix = HOST_PREFIXES[this.env];
+        const cloudPrefix = `${AICS_FMS_S3_URL_PREFIX}${AICS_FMS_S3_BUCKETS[this.env]}`;
+
+        if (this.path.startsWith(cloudPrefix)) {
+            const relativePath = this.path.replace(`${cloudPrefix}`, "");
+            return `${localPrefix}${relativePath}`;
         }
 
-        // AICS FMS files' paths are cloud paths
         return this.path;
```

Comment:
Is this `null` or the cloud path at this point? If `null` can you return `null` here explicitly & if it is the cloud path can you keep the comment above?

Reply:
This is for backwards compatibility. Right now its possible this.path is '/allen' in this case we want to return it.

Reply:
oh Im wrong. We already check in localPath for this. setting to null.

Reply:
resilved in [41c48db](https://github.com/AllenInstitute/biofile-finder/pull/389/commits/41c48db32edf9182ac477cd73ffb16de9f0a64f4)